### PR TITLE
Load Default SQLCmd Parameters value into dacpac

### LINKED ISSUE #248: SQLCMD Variable Defaults not Present in Built DacPac
We are using sql command variables to define the server and database on a project reference, but the default values are not making their way into the dacpacs when we do a `dotnet build`. We tried using the `dotnet build /p:SqlCmdVar__1=somevalue` but it didn't net a difference. Using v.1.17.3. 

Our sqlproj:
```
<Project Sdk="MSBuild.Sdk.SqlProj/1.17.3">
    <PropertyGroup>
        <TargetFramework>netstandard2.1</TargetFramework>
        <RecoveryMode>Simple</RecoveryMode>
        <SqlServerVersion>Sql130</SqlServerVersion>
    </PropertyGroup>
    <PropertyGroup>
        <OutputPath>deploy\</OutputPath>
        <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
        <AppendRuntimeIdentifierToOutputPath>false</AppendRuntimeIdentifierToOutputPath>
    </PropertyGroup>
    <ItemGroup>
        <Content Remove="scripts\**\*.sql" />
    </ItemGroup>
    <ItemGroup>
        <ProjectReference Include="../Soft_Landing/Soft_Landing.sqlproj" DatabaseVariableLiteralValue="Soft_Landing" />
        <ProjectReference Include="../Soft_App/Soft_App.sqlproj" DatabaseVariableLiteralValue="Soft_App" />
        <ProjectReference Include="../ACK/ACK.sqlproj" DatabaseSqlCmdVariable="ACK" ServerSqlCmdVariable="EXTServer" />
    </ItemGroup>
    <ItemGroup>
        <SqlCmdVariable Include="ACK">
            <DefaultValue>ACK</DefaultValue>
            <Value>$(SqlCmdVar__1)</Value>
        </SqlCmdVariable>
        <SqlCmdVariable Include="EXTServer">
            <DefaultValue>192.168.1.1</DefaultValue>
            <Value>$(SqlCmdVar__2)</Value>
        </SqlCmdVariable>
    </ItemGroup>
</Project>
```
Then when examining the model.xml of the dacpac:
```
...

		<CustomData Category="SqlCmdVariables" Type="SqlCmdVariable">
			<Metadata Name="ACK" Value="" />
			<Metadata Name="EXTServer" Value="" />
		</CustomData>
	</Header>
```

We are using Azure Data Studio to perform the SQL Compare from the built dacpacs but they fail due the values on these vars missing.

I also tried adding the explicit values in the variables at the top of the sqlproj to no avail.
```
<PropertyGroup>
      <SqlCmdVar__1>ACK</SqlCmdVar__1>
</PropertyGroup>
```

### PR DESCRIPTION
**Pull request by Geims83**
fixes #248

### CODE REVIEW

**Comment by ErikEJ** [comment]
You should be able to add a test, have a look at AddSqlCmdvariable test

**Comment by Geims83** [comment]
Tests added! Thank you for the hint.

**Review by ErikEJ — APPROVED** [review]
LGTM

**Review by jeffrosenberg — APPROVED** [review]

File: src/DacpacTool/PackageBuilder.cs [hunk 1]
```diff
@@ -50,17 +50,27 @@ private void ValidateReference(string referenceFile)
...
         }
 
-        public void AddSqlCmdVariables(string[] variableNames)
+        public void AddSqlCmdVariables(string[] variables)
         {
             // Ensure that the model has been created
             EnsureModelCreated();
 
-            foreach (var variableName in variableNames)
+            foreach (var variable in variables)
```
Comment (jmezach) [inline src/DacpacTool/PackageBuilder.cs:58]: Seems like this `foreach` loop only exists for logging purposes now, so we're essentially enumerating the list of variables twice. Perhaps we should move this logging into the `AddSqlCmdVariables()` method in `Extensions` instead?

**Review by jmezach — CHANGES_REQUESTED** [review]
Made one comment. Also, I was thinking we might want to add default values to the test project that uses SQLCMD variables as well.

File: src/DacpacTool/PackageBuilder.cs (hunk 1 above)
Comment (Geims83) [reply src/DacpacTool/PackageBuilder.cs:58]: What about keeping this loop and removing the one in `Extensions`, with a new `AddSqlCmdVariable(string Name, string DefaultValue)` method to 'Extensions'? Feels cleaner to me.

Comment (jmezach) [reply src/DacpacTool/PackageBuilder.cs:58]: Yeah, I agree that does feel cleaner.

**Comment by Geims83** [comment]
@jmezach fixed the issue in the comment.

> I was thinking we might want to add default values to the test project that uses SQLCMD variables as well.

It seems to me that there are already default values in sqlcmd variables in the projects where they are used. Am I missing something?

**Comment by jmezach** [comment]
Yes, you're right, sorry about that.

File: src/DacpacTool/Extensions.cs [hunk 2]
```diff
@@ -204,17 +204,13 @@ public static IEnumerable<string> GetReferencedDacPackages(this TSqlModel model)
...
             var dataSchemaModel = GetDataSchemaModel(model);
 
             var customData = Activator.CreateInstance(Type.GetType("Microsoft.Data.Tools.Schema.SchemaModel.CustomSchemaData, Microsoft.Data.Tools.Schema.Sql"), "SqlCmdVariables", "SqlCmdVariable");
-
-            foreach (var variableName in variableNames)
-            {
-                var setMetadataMethod = customData.GetType().GetMethod("SetMetadata", BindingFlags.Public | BindingFlags.Instance);
-                setMetadataMethod.Invoke(customData, new object[] { variableName, string.Empty });
-            }
+            var setMetadataMethod = customData.GetType().GetMethod("SetMetadata", BindingFlags.Public | BindingFlags.Instance);
```
Comment (jmezach) [inline src/DacpacTool/Extensions.cs:212]: Now that I look at this I'm wondering if this works the way we expect it to work. Previously we would only add one `CustomSchemaData` object to the model with all the variables. Now we're adding it multiple times for each variable. To be honest I don't know if that matters, but I think we should have a look at the resulting `.dacpac` from both a `.sqlproj` and an `MSBuild.Sdk.SqlProj` project to make sure we're producing the correct output.

Comment (Geims83) [reply src/DacpacTool/Extensions.cs]: Just found out issue #20 that was about this problem, so I reverted the commit and moved console logging inside `Extensions` as you were suggesting.

**Review by jmezach — APPROVED** [review]
LGTM

**Pull request merged**